File: 1232. Check If It Is a Straight Line - Easy/solution.cpp
static auto _ = [] () { ios_base::sync_with_stdio(false); cin.tie(nullptr); return 0; } ();
class Solution {
public:
    bool checkStraightLine(vector<vector<int>>& coordinates) {
        if (coordinates.size() < 3) {
            return true;
        }
        
        int A = coordinates[0][1]-coordinates[1][1];
        int B = coordinates[1][0]-coordinates[0][0];
        int C = - A*coordinates[0][0] - B*coordinates[0][1];
        
        for (int i = 2; i < coordinates.size(); ++i) {
            if (A * coordinates[i][0] + B * coordinates[i][1] + C != 0) {
                return false;
            }
        }
        
        return true;
    }
};

File: 1232. Check If It Is a Straight Line - Easy/solution.py
class Solution:
    def checkStraightLine(self, coordinates: List[List[int]]) -> bool:
        if len(coordinates) > 2:
            p0 = coordinates[0]
            p1 = coordinates[1]
            for p in coordinates[2:]:
                if p0[0] * (p1[1] - p[1]) + p1[0] * (p[1] - p0[1]) + p[0] * (p0[1] -p1[1]) != 0:
                    return False

        return True